extmod/modlwip.c

```diff
@@ -1316,7 +1316,7 @@ STATIC mp_obj_t lwip_socket_settimeout(mp_obj_t self_in, mp_obj_t timeout_in) {
         timeout = -1;
     } else {
         #if MICROPY_PY_BUILTINS_FLOAT
-        timeout = 1000 * mp_obj_get_float(timeout_in);
+        timeout = 1000 * (mp_uint_t)mp_obj_get_float(timeout_in);
```

Comment:
this needs to be a float multiplication, because the `timeout_in` may be less than one (to get a sub-second timeout value)